tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleShareConsumer.java

```
@@ -0,0 +1,186 @@
+        } finally {
+            consumerWrapper.cleanup();
+            opts.formatter().close();
+            reportRecordCount();
+
+            SHUTDOWN_LATCH.countDown();
+        }
+    }
+
+    private static void addShutdownHook(ConsumerWrapper consumer) {
+        Exit.addShutdownHook("consumer-shutdown-hook", () -> {
+            try {
+                consumer.wakeup();
+                SHUTDOWN_LATCH.await();
+            } catch (Throwable t) {
+                LOG.error("Exception while running shutdown hook: ", t);
+            }
+        });
+    }
+
+    static void process(int maxMessages, MessageFormatter formatter, ConsumerWrapper consumer, PrintStream output,
+                        boolean rejectMessageOnError, AcknowledgeType acknowledgeType) {
+        while (messageCount < maxMessages || maxMessages == -1) {
+            ConsumerRecord<byte[], byte[]> msg;
+            try {
+                msg = consumer.receive();
+            } catch (WakeupException we) {
+                LOG.trace("Caught WakeupException because consumer is shutdown, ignore and terminate.");
+                // Consumer will be closed
+                return;
+            } catch (Throwable t) {
+                LOG.error("Error processing message, terminating consumer process: ", t);
+                // Consumer will be closed
+                return;
+            }
+            messageCount += 1;
+            try {
+                formatter.writeTo(new ConsumerRecord<>(msg.topic(), msg.partition(), msg.offset(), msg.timestamp(), msg.timestampType(),
+                        0, 0, msg.key(), msg.value(), msg.headers(), Optional.empty()), output);
+                consumer.acknowledge(msg, acknowledgeType);
+            } catch (Throwable t) {
+                if (rejectMessageOnError) {
+                    LOG.error("Error processing message, skipping this message: ", t);
```

This error message still needs changing to say "rejecting" rather than "skipping".

---

yes, had missed this, have updated now. Thanks.